### PR TITLE
Customclass warning

### LINKED ISSUE #1348: Better error messages for wrongly define slot type
**Rasa Core version**:
0.12.x
**Python version**: 
2.7.15
**Operating system** (windows, osx, ...):
Ubuntu (Docker)
**Issue**:
When training my model I get the following error:
```
Traceback (most recent call last):
  File "/opt/conda/envs/env/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/opt/conda/envs/env/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/opt/conda/envs/env/lib/python2.7/site-packages/rasa_core/train.py", line 361, in <module>
    additional_args)
  File "/opt/conda/envs/env/lib/python2.7/site-packages/rasa_core/train.py", line 265, in do_default_training
    kwargs=additional_arguments)
  File "/opt/conda/envs/env/lib/python2.7/site-packages/rasa_core/train.py", line 172, in train_dialogue_model
    policies=policies)
  File "/opt/conda/envs/env/lib/python2.7/site-packages/rasa_core/agent.py", line 202, in __init__
    self.domain = self._create_domain(domain)
  File "/opt/conda/envs/env/lib/python2.7/site-packages/rasa_core/agent.py", line 652, in _create_domain
    return Domain.load(domain)
  File "/opt/conda/envs/env/lib/python2.7/site-packages/rasa_core/domain.py", line 93, in load
    return cls.from_yaml(read_file(filename))
  File "/opt/conda/envs/env/lib/python2.7/site-packages/rasa_core/domain.py", line 99, in from_yaml
    return cls.from_dict(data)
  File "/opt/conda/envs/env/lib/python2.7/site-packages/rasa_core/domain.py", line 104, in from_dict
    slots = cls.collect_slots(data.get("slots", {}))
  File "/opt/conda/envs/env/lib/python2.7/site-packages/rasa_core/domain.py", line 191, in collect_slots
    slot = slot_class(slot_name, **slot_dict[slot_name])
TypeError: 'NoneType' object is not callable
```

**Content of domain file** (if used & relevant):
```yaml
slots:
  auth.access_token:
    type: unfeaturized
  auth.login_type:
    type: unfeaturized
  auth.authenticated:
    type: boolean
    initial_value: false
  auth.username:
    type: unfeaturized
  auth.password:
    type: unfeaturized
  name:
    type: unfeaturized
    initial_value: ""
  shopping_item:
    type: list
  attribute:
    type: list
  products:
    type: list
  user_id:
    type: text
  next:
    type: text

```

### PR DESCRIPTION
**Pull request by imLew**
**Proposed changes**:
- fixes #1348

The problem was that if provided an incorrect name for a slot rasa would try to import it as a custom slot type
using `class_from_module_path` which returned `None` if it  could not find the class.
This has been changed to throw an `ImportError`.
Furthermore this function is called in several places, two of them have been updated to except and throw more specific exceptions instead.

**Status (please check what you already did)**:
- [x] made PR ready for code review

### CODE REVIEW

**Comment by akelad** [comment]
@imLew could you fix whatever is causing the build to fail please?

File: rasa_core/policies/ensemble.py [hunk 1]
```diff
@@ -249,12 +249,11 @@ def from_dict(cls, dictionary: Dict[Text, Any]) -> List[Policy]:
...
 
-            constr_func = utils.class_from_module_path(policy_name)
-
-            if constr_func:
+            try:
+                constr_func = utils.class_from_module_path(policy_name)
                 policy_object = constr_func(**policy)
                 parsed_policies.append(policy_object)
-            else:
+            except(ImportError):
```
Comment (akelad) [inline rasa_core/policies/ensemble.py]: this also needs to accept an AttributeError, did you try the case where you specify the policy as "test.TestPolicy"? maybe write a test for this as well if one doesn't exist yet

File: rasa_core/slots.py [hunk 2]
```diff
@@ -63,17 +63,16 @@ def __repr__(self):
...
     def resolve_by_type(type_name):
         """Returns a slots class by its type name."""
-
         for cls in utils.all_subclasses(Slot):
             if cls.type_name == type_name:
                 return cls
         try:
             return utils.class_from_module_path(type_name)
-        except Exception:
+        except(ImportError):
```
Comment (akelad) [inline rasa_core/slots.py]: same here with the AttributeError

File: rasa_core/policies/ensemble.py (hunk 1 above)
Comment (imLew) [reply rasa_core/policies/ensemble.py]: Not quite sure what you mean as I don't know the workings of loading policies yet.
How would this be different from the `test_invalid_policy_configurations` that we already have?

**Pull request merged**